packages/ui/components/Heartbeat.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <div class="countdown">
+    <div v-if="!timeOver">{{ hours }}:{{ minutes }}:{{ seconds }}</div>
+    <div v-else>00:00:00</div>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    milliseconds: {
+      type: String,
+      required: true,
+    },
+    lastResultTimestamp: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      now: new Date().getTime(),
+      polling: null,
+      timeOver: false,
+    }
+  },
+  computed: {
+    distance() {
+      return this.countdown - this.now
+    },
+    countdown() {
+      // Last result timestamp to milliseconds plus heartbeat
+      return (
+        Number(`${this.lastResultTimestamp}000`) + Number(this.milliseconds)
+      )
+    },
+    second() {
+      return 1000
+    },
+    minute() {
+      return this.second * 60
+    },
+    hour() {
+      return this.minute * 60
+    },
+    day() {
+      return this.hour * 24
+    },
+    hours() {
+      const hours = Math.floor((this.distance % this.day) / this.hour)
+      return hours < 10 ? `0${hours}` : hours
+    },
+    minutes() {
+      const minutes = Math.floor((this.distance % this.hour) / this.minute)
+      return minutes < 10 ? `0${minutes}` : minutes
+    },
+    seconds() {
+      const seconds = Math.floor((this.distance % this.minute) / this.second)
+      return seconds < 10 ? `0${seconds}` : seconds
+    },
+  },
+  watch: {
+    distance(value) {
+      if (value < 0) {
+        clearInterval(this.polling)
+        this.timeOver = true
```

Comment:
How can it be restarted when a new transaction is found?

Reply:
the last transaction result timestamp is reactive